src/commands/exile.py

```diff
@@ -0,0 +1,53 @@
+import discord
+from discord.ext.commands import Bot
+from settings import get_settings
+from services.exile_service import exile_user, unexile_user
+from collections.abc import Coroutine
+
+settings = get_settings()
+
+def create_unexile_command(bot: Bot) -> None:
+    @bot.tree.command()
+    @discord.app_commands.describe(
+        user="User being exiled"
+    )
+    async def unexile(interaction: discord.Interaction, user: discord.Member):
+        """Exile the specified user."""
+
+        await run_command_with_logging(interaction, unexile_user, user)
+
+        await interaction.response.send_message(f"Successfully unexiled {user.mention}", ephemeral=True)
+
+def create_exile_command(bot: Bot) -> None:
+    @bot.tree.command()
+    @discord.app_commands.describe(
+        user="User being exiled",
+        duration="The duration of the exile. TBA format",
+        reason="Reason for exile"
+    )
+    async def exile(interaction: discord.Interaction, user: discord.Member, duration: str, reason: str):
+        """Exile the specified user."""
+
+        await run_command_with_logging(interaction, exile_user, user, duration, reason)
+
+        await interaction.response.send_message(f"Successfully exiled {user.mention}", ephemeral=True)
+
+
+# NB this feels very messy, fragile, and easy to mess up
+# There might be a better way to wrap all this in something cleaner and better reusable
+async def run_command_with_logging(interaction, command: Coroutine, *args):
```

Comment:
not sure if we would also want to write to the logging buffer in this function as well. imo we should use `log_info_and_embed` for the error messages/handling and then use `run_command_with_logging` for the successful operations

Reply:
I've updated `log_info_and_embed` and `run_command_with_logging` to try and make it a bit more clear the purposes of each, the intent is to have `log_info_and_embed` be run any time you want to write a single log message, and `run_command_with_logging` be a wrapper around the command's business logic to handle the creation of the embed, and the sending of that embed to the logging discord channel for you